signature/sign.go

```diff
@@ -129,6 +181,21 @@ func Verify(s *pipeline.Signature, keySet jwk.Set, env map[string]string, sf Sig
 		return err
 	}
 
+	for it := keySet.Keys(context.Background()); it.Next(context.Background()); {
```

Comment:
I was hacking this together and so left this as `context.Background` but we should probably now make that an option too

Reply:
Should we pass in a ctx @patrobinson?

Reply:
Yeah we should allow passing one in, agent already has one we will want to use